### PR TITLE
cpio: reuse patch for %gcc10: for %clang as well

### PR DESCRIPTION
**Pull request by luszczek**
discoverd in PR #26517

### CODE REVIEW

**Review by scheibelp — DISMISSED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/cpio/package.py [hunk 1]
```diff
@@ -19,7 +19,8 @@ class Cpio(AutotoolsPackage, GNUMirrorPackage):
 
     version('2.13', sha256='e87470d9c984317f658567c03bfefb6b0c829ff17dbf6b0de48d71a4c8f3db88')
 
-    patch('https://src.fedoraproject.org/rpms/cpio/raw/dfe64c466d3ea2c8dfbd99700d9006f610064167/f/cpio-2.13-mutiple-definition.patch', sha256='d22633c368b8aedf4c08b23b6fbaa81a52404c8943ab04926404083ac10f1a4b', when='%gcc@10:')
+    for ccomp in ('%gcc@10:', '%clang'):
+        patch('https://src.fedoraproject.org/rpms/cpio/raw/dfe64c466d3ea2c8dfbd99700d9006f610064167/f/cpio-2.13-mutiple-definition.patch', sha256='d22633c368b8aedf4c08b23b6fbaa81a52404c8943ab04926404083ac10f1a4b', when=ccomp)
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/cpio/package.py]: That line is really long, and the CI failed due to raja marked as failing, so we've to re-run the the GitLab CI anyways.

What about reformatting it like this and then, rebasing it to the current develop branch which has the fix for raja:
```suggestion
        patch('https://src.fedoraproject.org/rpms/cpio/raw/dfe64c466d3ea2c8dfbd99700d9006f610064167/f/cpio-2.13-mutiple-definition.patch', sha256='d22633c368b8aedf4c08b23b6fbaa81a52404c8943ab04926404083ac10f1a4b', when=ccomp)
              sha256='d22633c368b8aedf4c08b23b6fbaa81a52404c8943ab04926404083ac10f1a4b',
              when=ccomp)
```
A new commit (in that case rebased to the latest develop with raja fixed) fixed GitLab CI failing due to raja in this PR: #27167

**Comment by bernhardkaindl** [comment]
If the new commit does not cause CI to succeed, you have to rebase to the current develop:

The GitHub CLI tool helps to clone a PR for fixing it up: https://cli.github.com/ (or `spack install gh`), then:
```
git checkout develop
git pull
gh co 27589
git rebase develop
git push -f
```
(you might have to add the remote of the PR though and then specify it on the push)

**Comment by scheibelp** [comment]
Close/reopen to fix strange style check error.

**Comment by bernhardkaindl** [comment]
The style error was due to an oversight by me causing an actual syntax error when editing the change by manual edit thru the github review UI.

Update: Simplified the fix for multiple definitions and to not have to download a patch using filter_file (because it's really trivial to see).

This makes the change more transparent in the review of the `package.py` file,
obsoletes having to download and cache the patch file,
and avoids the dependency on requiring that patch file to be downloadable at build time.

Rebased to current develop as well, which should fix the raja issue in the GitLab CI.

**Review by bernhardkaindl — APPROVED** [review]

**Pull request merged**